Network-Office/src/main/java/dev/office/networkoffice/gathering/service/GatheringApplyService.java

```diff
@@ -0,0 +1,92 @@
+package dev.office.networkoffice.gathering.service;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import dev.office.networkoffice.gathering.entity.Applicants;
+import dev.office.networkoffice.gathering.entity.Gathering;
+import dev.office.networkoffice.gathering.repository.ApplicantRepository;
+import dev.office.networkoffice.gathering.repository.GatheringRepository;
+import dev.office.networkoffice.user.entity.User;
+import dev.office.networkoffice.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class GatheringApplyService {
+
+	private final ApplicantRepository applicantRepository;
+	private final GatheringRepository gatheringRepository;
+	private final UserRepository userRepository;
+	private final GatheringAuthorityManagerService managerService;
+
+	/**
+	 * 모임 신청.
+	 * @param applicantId : 신청자 아이디
+	 * @param gatheringId : 신청하려는 모임 아이디
+	 */
+	@Transactional
+	public void applyGathering(Long applicantId, Long gatheringId){
+		User user = userRepository.findById(applicantId)
+			.orElseThrow(()-> new RuntimeException("유저없음"));//신청자 검증
+		Gathering gathering = gatheringRepository.findById(gatheringId)
+			.orElseThrow(()-> new RuntimeException("모임없음")); //모임검증
+		//기존 모임에 이미 user가 있는지 체크
+
+		//신청 엔티티 생성
+		Applicants applicants = Applicants.builder()
+			.gathering(gathering)
+			.user(user)
+			.build();
+		applicantRepository.save(applicants);
+	}
+
+	/**
+	 * 호스트가 볼 수 있는 모임 신청자들 목록
+	 * @param hostId
+	 * @param gatheringId
+	 */
+
+	@Transactional(readOnly = true)
+	public List<Applicants> readApplicantsByGathering(Long hostId, Long gatheringId){
+		Gathering gathering = gatheringRepository.findById(gatheringId)
+			.orElseThrow(()-> new RuntimeException("모임없음"));
+		if ((gathering.getHostId().equals(hostId))){ //해당 유저가 호스트인지 검사한다.
+			throw new RuntimeException("호스트가 아니라 조회 할 수 없다.");
+		}
+		return applicantRepository.findAllByGathering(gathering);
+	}
```

Comment:
```suggestion
	@Transactional(readOnly = true)
	public List<Applicants> readApplicantsByGathering(Long hostId, Long gatheringId){
	    // ApplicantRepository에 메서드 추가 필요
	    return applicantRepository.findByIdAndGatheringHostId(applicantId, hostId)
				.orElseThrow(() -> new IllegalArgumentException("신청id로 찾음"));
	}
```

- Applicant 엔티티를 조회하고 내부 내용을 확인하는 로직 대신 한번의 쿼리로 조건에 맞는 엔티티를 반환하거나 예외를 처리하는 것은 어떨까요?
- 추가로 `RuntimeException` 보다는 `IllegalArgumentException`이 더 어울리는 것 같아요. 
(잘못된 id로 요청이 들어오는 경우는 예를 들어 동기화가 되지 않은 화면에서 요청을 전송하는 경우가 있을 것 같아요.)

Reply:
좋은 생각입니다! 수정해보겠습니다.☺️